vms/platformvm/state/state.go

```diff
@@ -117,6 +117,9 @@ type Chain interface {
 	AddSubnetTransformation(transformSubnetTx *txs.Tx)
 
 	AddChain(createChainTx *txs.Tx)
+	GetChainSubnet(chainID ids.ID) (ids.ID, error)
```

Comment:
to be fair, `GetChainSubnet` is not really called in tx execution, so we don't strictly need it here.
However introducing it has the following advantages:

1. It gives a clear indication on what kind of index we should support on a merkleDB based P-chain. Such indexes can be local ones as they are used for APIs (not in tx execution)
2. It introduces a getter for chains index.